br/pkg/streamhelper/advancer.go

```diff
@@ -470,8 +470,10 @@ func (c *CheckpointAdvancer) onTaskEvent(ctx context.Context, e TaskEvent) error
 func (c *CheckpointAdvancer) setCheckpoint(s spans.Valued) bool {
 	cp := newCheckpointWithSpan(s)
 	if cp.TS < c.lastCheckpoint.TS {
-		log.Warn("failed to update global checkpoint: stale",
-			zap.Uint64("old", c.lastCheckpoint.TS), zap.Uint64("new", cp.TS))
+		if cp.TS != 0 {
```

Comment:
what's the purpose of the check?

Reply:
good question, in the log I see there is a bunch of false alarm saying failed to update global checkpoint with new cp ts being 0. after debugging I realized it's a corner case at the initialization step that a 0 can be in the queue but it won't affect anything, just producing a bunch of noisy logs. 
I tried to fix the corner case but it's more complex than I thought and need to touch critical code and I don't feel like it worths the effort and risk, so just to prevent the spam logging at here.

cc @YuJuncen since you are the main contributor of this logic. Let me know if you have a better fix! thanks